jobsdb/admin.go

```diff
@@ -136,3 +140,96 @@ func (jd *HandleT) failExecutingDSInTx(txHandler transactionHandler, ds dataSetT
 	)
 	return err
 }
+
+func (jd *HandleT) startCleanupLoop(ctx context.Context) {
+	jd.backgroundGroup.Go(misc.WithBugsnag(func() error {
+		jd.oldJobsCleanupRoutine(ctx)
+		return nil
+	}))
+}
+
+func (jd *HandleT) oldJobsCleanupRoutine(ctx context.Context) {
+	jd.doCleanupOldJobs(ctx)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-jd.TriggerJobCleanUp():
+			jd.doCleanupOldJobs(ctx)
+		}
+	}
+}
+
+func (jd *HandleT) doCleanupOldJobs(ctx context.Context) {
+	tags := statTags{CustomValFilters: []string{jd.tablePrefix}}
+	command := func() interface{} {
+		return jd.WithUpdateSafeTx(ctx, func(tx UpdateSafeTx) error {
+			dsList := jd.getDSList()
```

Comment:
Querying by a batch of 1000 won't necessarily put less stress on the database. This would help in the case where we find 1000 jobs and stop querying further. But in most cases, we wouldn't have any jobs at all that need to be cleaned. Meaning we'd keep querying all the dataSets. Consider the case you mention, where there's a lot of Datasets, we'd be unnecessarily querying all of them even though there's no jobs that require cleanup at all.
To counter this I've introduced another filter to limit the number of dataSets queried which should be sufficient imo.

I can think of passing some `Opts`(to limit the number of DSs) to `GetUnprocessed` and `GetProcessed` to get the best of both worlds.

Reply:
No we shouldn't query all the datasets if the first batch doesn't contain any jobs to be cleaned up. Jobs are returned in ascending order.